src/main/java/com/ericsson/ei/controller/RuleCheckControllerImpl.java

```diff
@@ -72,32 +73,35 @@
 
     @Autowired
     private IRuleCheckService ruleCheckService;
-    
+
     @Setter
     @Value("${testaggregated.enabled:false}")
     private Boolean testEnable;
 
     /**
      * This method interacts with JmesPathInterface class method runRuleOnEvent to
      * evaluate a rule on JSON object.
-     * 
+     *
      * @param rule-
      *            takes a String as a rule that need to be evaluated on JSON content
      * @param jsonContent-
      *            takes JSON object as a String
      * @return a String object
-     * 
+     *
      */
     @Override
     @CrossOrigin
     @ApiOperation(value = "To execute rule on JSON", response = String.class)
     public ResponseEntity<?> updateRulesRuleCheck(
-            @ApiParam(value = "JMESPath rule", required = true) @RequestParam String rule,
-            @ApiParam(value = "JSON object", required = true) @RequestBody String jsonContent) {
+            @ApiParam(value = "JSON object", required = true) @RequestBody RuleCheckBody body) {
+        JSONObject rule = new JSONObject(body.getRule().getAdditionalProperties());
+        JSONObject event = new JSONObject(body.getEvent().getAdditionalProperties());
+
+        String ruleString = rule.toString().replaceAll("\"", "");
```

Comment:
Doesn't these JSON objects has a function that is called "asText()", I think JsonNode object has this function anyway , which removes these extra quotations. That function could be used, if I now understand this situation correct.

Reply:
The reason for the replacement is that jmespath.compile(rule) that is executed is very sensitive and doesn't allow quotations in the expression. asText() sometimes makes empty texts and i don't really know why.